### PR TITLE
Adds MGF file validation on import

### PR DESCRIPTION
**Pull request by ElDeveloper**
Checks for unique MS1 ids and at least one MS2 per MS1.

### CODE REVIEW

File: q2_qemistree/_semantics.py [hunk 1]
```diff
@@ -12,10 +12,61 @@
...
+        if line.startswith('FEATURE_ID='):
+            # get the feature identifier without the new line
+            new_feature_id = line.split('=')[1].strip()
+
+            if read_id is None:
+                read_id = new_feature_id
+            else:
+                if new_feature_id != read_id:
+                    if ms2 < 1:
+                        raise ValueError('Feature "%s" does not have at least '
```
Comment (anupriyatripathi) [inline q2_qemistree/_semantics.py:31]: Can we replace this by "At least one feature (Feature ID = XYZ) does not have MS2 information" & make this a warning?

File: q2_qemistree/_semantics.py [hunk 2]
```diff
@@ -12,10 +12,61 @@
...
+                                         'one MSLEVEL=2 record' % read_id)
+                    if new_feature_id in observed_ids:
+                        raise ValueError('Feature "%s" has repeated records, '
+                                         'all records for each feature are '
+                                         'assumed to be listed contiguously.'
+                                         % new_feature_id)
```
Comment (anupriyatripathi) [inline q2_qemistree/_semantics.py:37]: I think we can get rid of this. As Marcus said, SIRIUS ignores this case.

File: q2_qemistree/_semantics.py [hunk 3]
```diff
@@ -12,10 +12,61 @@
...
+
+                    ms1, ms2 = 0, 0
+
+        elif read_id is not None:
+            if line.startswith('MSLEVEL=1'):
+                ms1 += 1
+
+                if ms1 > 1:
+                    raise ValueError('Feature "%s" has more than one MSLEVEL=1'
+                                     ' record' % read_id)
```
Comment (anupriyatripathi) [inline q2_qemistree/_semantics.py:52]: This is also OK per Marcus

Comment (ElDeveloper) [reply q2_qemistree/_semantics.py:47]: Excellent!

**Comment by ElDeveloper** [comment]
Thanks so much!

**Pull request merged**